src/aiida_quantumespresso/tools/calculations/pw.py

```diff
@@ -92,3 +98,80 @@ def get_magnetic_configuration(self, atol: float = 0.5, ztol: float = 0.05) -> A
         return AttributeDict(
             {'structure': structure, 'magnetic_moments': None if non_magnetic else results['magnetic_moments']}
         )
+
+    def get_occupations(self, reshape=False) -> list[dict]:
+        """Return the occupations for a PwCalculation/PwBaseWorkChain node as a list of python dictionaries."""
+
+        # assert first that this is a Hubbard calculation
+        # so for instance if you try to call this method on a normal pw calculation it raises a ValueError
+        if not isinstance(self._node.inputs.structure, HubbardStructureData):
+             raise ValueError('The input structure is not a HubbardStructureData node.')
+
+        
+        with self._node.outputs.retrieved.base.repository.open('data-file-schema.xml') as xml_file:
+            xml_parsed = ElementTree.parse(xml_file)
+
+        schema_filepath = get_schema_filepath(xml_parsed)
+        xsd = XMLSchema(schema_filepath)
+        xml_dictionary = xsd.to_dict(xml_parsed, validation='skip')
+
+        dft_u = xml_dictionary['output']['dft']['dftU']
+        
+        # We aggregate data by atom index
+        aggregated_data = {}
+
+        # Helper to standardize processing. 
+        # We treat the two lists differently based on your specifications.
+        # 1. Hubbard_ns_nc: Non-collinear (Single matrix, ignore @spin)
+        # 2. Hubbard_ns:    Collinear (Split by @spin) OR No-Polarization (Single matrix, no @spin)
+        
+        source_lists = [
+            (dft_u.get('Hubbard_ns_nc', []), True),  # (List, is_non_collinear)
+            (dft_u.get('Hubbard_ns', []),    False)  # (List, is_standard_ns)
+        ]
+
+        # this needs to be verified together with the QE developers, but it seems that in the case of non-collinear
+        # calculations, what is reported as "occupation_matrix" is actually the absolute value,
+        # since it should have both real and imaginary parts
+
+        for entries, is_non_collinear in source_lists:
+            for entry in entries:
+                atom_index = entry['@index']
+                
+                # Parse matrix
+                occ_matrix = np.array(entry['$'])
+                if reshape:
+                    occ_matrix = occ_matrix.reshape(entry['@dims'])
+
+                # Initialize container if new
+                if atom_index not in aggregated_data:
+                    aggregated_data[atom_index] = {
+                        'atom_label': atom_index,
+                        'atom_specie': entry['@specie'],
+                        'shell': entry['@label'],
+                        'occupation_matrix': {} 
```

Comment:
i am wondering if it was worth it to just name it `occupations`. what do you think? @mbercx any strong opinion?